cdap-cli/src/main/java/co/cask/cdap/cli/CLIConfig.java

```diff
@@ -155,25 +180,30 @@ private AccessToken acquireAccessToken(ClientConfig clientConfig, ConnectionConf
     }
 
     try {
-      AccessToken savedAccessToken = getSavedAccessToken(connectionInfo.getHostname());
-      checkConnection(clientConfig, connectionInfo, savedAccessToken);
-      return savedAccessToken;
+      UserAccessToken savedToken = getSavedAccessToken(connectionInfo.getHostname());
+      connectionInfo = new AuthenticatedConnectionConfig(connectionInfo, savedToken.getUsername());
+      clientConfig.setConnectionConfig(connectionInfo);
```

Comment:
Don't set connection config here. This method is only intended to acquire an access token, and can't assume that the process of using the new connection will run without failure after this method is executed.